### PR TITLE
GPP support

### LINKED ISSUE #692: GPP Support
**Is your feature request related to a problem? Please describe.**
The IAB has defined a new Global Privacy Platform. Prebid's handling of this platform is described at https://docs.google.com/document/d/1dRxFUFmhh2jGanzGZvfkK_6jtHPpHXWD7Qsi6KEugeE/edit#

**Describe the solution you'd like**
Both iOS and Android SDKs 

```
Read the IABGPP_GppString value from NSUserDefaults(iOS)/SharedPreferences(Android)  and pass to PBS as regs.gpp.
```

### PR DESCRIPTION
**Pull request by OlenaPostindustria**
Closes #692

### CODE REVIEW

File: PrebidMobile/ConfigurationAndTargeting/UserConsentDataManager.swift [hunk 1]
```diff
@@ -61,6 +65,16 @@ public class UserConsentDataManager: NSObject {
         UserDefaults.standard.getObjectFromUserDefaults(forKey: IABUSPrivacy_StringKey)
     }
     
+    // MARK: - GPP
+    
+    public var gppHDRString: String? {
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/ConfigurationAndTargeting/UserConsentDataManager.swift:70]: General question about `UserConsentDataManager` visibility. Can we make this class internal for the SDK and not visible for SDK users?

Comment (OlenaPostindustria) [reply PrebidMobile/ConfigurationAndTargeting/UserConsentDataManager.swift:70]: Sure, we can make it internal

File: PrebidMobile/ConfigurationAndTargeting/Targeting.swift [hunk 2]
```diff
@@ -151,7 +151,24 @@ public class Targeting: NSObject {
         get { UserConsentDataManager.shared.subjectToCOPPA }
     }
     
+    // MARK: - СCPA
+    
+    public var usPrivacyString: String? {
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/ConfigurationAndTargeting/Targeting.swift:156]: we shouldn't provide an API way for CCPA properties

File: PrebidMobile/ConfigurationAndTargeting/UserConsentDataManager.swift [hunk 3]
```diff
@@ -51,16 +55,26 @@ public class UserConsentDataManager: NSObject {
      Integer flag indicating if this request is subject to the COPPA regulations
      established by the USA FTC, where 0 = no, 1 = yes
      */
-    public var subjectToCOPPA: Bool? {
+    var subjectToCOPPA: Bool? {
         set { UserDefaults.standard.set(newValue, forKey: PB_COPPAKey) }
         get { UserDefaults.standard.object(forKey: PB_COPPAKey) != nil ? UserDefaults.standard.bool(forKey: PB_COPPAKey) : nil }
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/ConfigurationAndTargeting/UserConsentDataManager.swift:60]: need to remove writing this value into UserDefaults. Use the same approach as for the TCF signal here as well.

File: PrebidMobile/ConfigurationAndTargeting/Targeting.swift [hunk 4]
```diff
@@ -151,7 +151,24 @@ public class Targeting: NSObject {
...
+        UserConsentDataManager.shared.usPrivacyString
+    }
+    
+    // MARK: - GPP
+    
+    public var gppHDRString: String? {
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/ConfigurationAndTargeting/Targeting.swift:162]: we shouldn't provide an API way for GPP properties

**Review by YuriyVelichkoPI — CHANGES_REQUESTED** [review]

**Review by YuriyVelichkoPI — APPROVED** [review]

**Comment by ankit-thanekar007** [comment]
@YuriyVelichkoPI @OlenaPostindustria In the current release 2.2.0, I don't see these changes. Were they missed or is there a reason for them being not in release ? 

https://github.com/prebid/prebid-mobile-ios/blob/7034d236f71cc2dc9b506a567280be3e83836de8/PrebidMobile/PrebidMobileRendering/Networking/Parameters/PBMBasicParameterBuilder.m#L130

**Pull request merged**